meshtastic/interfaces/ble/lifecycle_primitives.py

```
@@ -0,0 +1,464 @@
+                    "Error running safe_cleanup hook for %s",
+                    operation_name,
+                    exc_info=True,
+                )
+                if cleanup_ran:
+                    return
+        try:
+            cleanup()
+        except Exception:  # noqa: BLE001 - shutdown cleanup must remain best effort
+            logger.debug("Error during %s", operation_name, exc_info=True)
+
+    def safe_execute(
```

Ruff (0.15.6)

[warning] Too many return statements (10 > 6)

(PLR0911)